go/pkg/client/bytestream.go

```diff
@@ -26,33 +48,52 @@ func (c *Client) WriteBytes(ctx context.Context, name string, data []byte) error
 	return err
 }
 
+// WriteBytesWithOptions uploads a byte slice with a given resource name to the CAS
+// at an arbitrary offset but retries still resend from the specified offset.
+func (c *Client) WriteBytesWithOptions(ctx context.Context, name string, data []byte, opts ...ByteStreamWriteOption) (int64, error) {
```

Comment:
Can you just pass in the offset and finishWrite here? I think that would be simpler since its only two args.

Reply:
In case if we add more options to the method(or to writeChunked method) in the future and this method keeps a variadic option, we can add them easily without affecting the call sites of this method. I'm not sure how likely we will add more options but I think that scales better. What do you think?

Reply:
Resolved offline. Updated the method signatures.